src/client/conn.rs

```diff
@@ -238,8 +238,32 @@ where
         }
     }
 
-    //TODO: replace with `impl Future` when stable
-    pub(crate) fn send_request_retryable(&mut self, req: Request<B>) -> Box<Future<Item=Response<Body>, Error=(::Error, Option<Request<B>>)> + Send>
+    #[cfg(impl_trait_available)]
+    pub(crate) fn send_request_retryable(&mut self, req: Request<B>) -> impl Future<Item = Response<Body>, Error = (::Error, Option<Request<B>>)> + Send
```

Comment:
Oh neat! I thought that even with `cfg`s, the older compiler would fail at the parsing step (which happens before `cfg`s are applied).

Reply:
It's not a requirement, I'm just thinking out loud here: do you know if there's a way that the code inside these two functions could only be written once, so that edits to one doesn't mean the other accidentally drifts?

Reply:
Type-position macro?

```rust
#[cfg(impl_trait_available)]
macro_rules! impl_trait_or_box {
    (ty: $($t:tt)+) => {
        impl $($t)+
    };
    (expr: $e:expr) => {
        $e
    };
}

#[cfg(not(impl_trait_available))]
macro_rules! impl_trait_or_box {
    (ty: $($t:tt)+) => {
        Box<$($t)+>
    };
    (expr: $e:expr) => {
        Box::new($e)
    };
}

fn method(...) -> impl_trait_or_box!(ty: Future + Send) {
    ...

    impl_trait_or_box!(expr: res)
}
```